src/org/traccar/protocol/Arnavi4ProtocolDecoder.java

```diff
@@ -0,0 +1,170 @@
+
+                default:
+                    break; // Skip other tags
+            }
+        }
+
+        return position;
+    }
+
+    @Override
+    protected Object decode(
+            Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+
+        ChannelBuffer buf = (ChannelBuffer) msg;
+
+        byte startSign = buf.readByte();
+
+        if (startSign == HEADER_START_SIGN) {
+
+            byte version = buf.readByte();
+
+            String imei = String.valueOf(buf.readLong());
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, imei);
+
+            if (deviceSession != null && channel != null) {
+
+                final ChannelBuffer response;
+
+                if (version == HEADER_VERSION_1) {
+                    response = ChannelBuffers.dynamicBuffer(ByteOrder.LITTLE_ENDIAN, 4);
+                    response.writeBytes(new byte[]{0x7B, 0x00, 0x00, 0x7D});
+
+                } else if (version == HEADER_VERSION_2) {
+                    response = ChannelBuffers.dynamicBuffer(ByteOrder.LITTLE_ENDIAN, 9);
+                    response.writeBytes(new byte[]{0x7B, 0x04, 0x00});
+                    byte[] timeBytes = ByteBuffer.allocate(4).putInt((int) (System.currentTimeMillis() / 1000)).array();
+                    response.writeByte(Checksum.modulo256(timeBytes));
+                    response.writeBytes(timeBytes);
+                    response.writeByte(0x7D);
+
+                } else {
+                    throw new IllegalArgumentException("unsupported header version");
+                }
+
+                channel.write(response);
+            }
+
+            return null;
+        }
+
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress);
+        if (deviceSession == null) {
+            return null;
+        }
+
+        if (startSign == PACKAGE_START_SIGN) {
+
+            List<Position> positions = new LinkedList<>();
+
+            int parcelNumber = buf.readUnsignedByte();
+
+            byte recordStartSign = buf.readByte();
+            while (recordStartSign != PACKAGE_END_SIGN) {
+                switch (recordStartSign) {
+                    case RECORD_PING:
+                    case RECORD_DATA:
+                    case RECORD_TEXT:
+                    case RECORD_FILE:
+                    case RECORD_BINARY:
+                        int length = buf.readUnsignedShort();
+                        long timestamp = buf.readUnsignedInt() * 1000;
+                        ChannelBuffer recordBuf = buf.readBytes(length);
+
+                        if (recordStartSign == RECORD_DATA) {
+                            positions.add(decodePosition(deviceSession, recordBuf, timestamp));
+                        }
+
+                        buf.readUnsignedByte(); // crc
+                        break;
+
+                    default:
+                        throw new IllegalArgumentException("unsupported record type");
```

Comment:
We can't just ingore it, because we won't know how much bytes to read. Removed throwing and added `return null`.